lib/Parse/ParseDecl.cpp

```diff
@@ -4808,7 +4808,7 @@ Parser::parseDeclVar(ParseDeclOptions Flags,
     } else if (Flags.contains(PD_InStruct) || Flags.contains(PD_InEnum) ||
                Flags.contains(PD_InProtocol)) {
       if (StaticSpelling == StaticSpellingKind::KeywordClass)
-        diagnose(Tok, diag::class_var_not_in_class)
+        diagnose(Tok, Flags.contains(PD_InProtocol) ? diag::class_var_not_in_class_protocol : diag::class_var_not_in_class)
```

Comment:
Nit: This line should be wrapped to 80 characters.  You can pull out diagnostic selectors as variables or parametrize the diagnostic.  Either one would shorten the line.